server/bin/pbench-backup-tarballs.py

```diff
@@ -383,39 +364,45 @@ def backup_data(s3_obj, config, logger):
                    ns3_fail=ns3_fail,
                    nquaran=nquaran)
 
-
-def main(parsed):
-    ret_status = 0
-    if not parsed.cfg_name:
+def main():
+    cfg_name = os.environ.get("CONFIG")
+    if not cfg_name:
         print("{}: ERROR: No config file specified; set CONFIG env variable or"
               " use --config <file> on the command line".format(_NAME_),
               file=sys.stderr)
         return 2
 
     try:
-        config = PbenchConfig(parsed.cfg_name)
+        config = PbenchConfig(cfg_name)
     except BadConfig as e:
         print("{}: {}".format(_NAME_, e), file=sys.stderr)
         return 1
 
-    global _logger
-    _logger = get_pbench_logger(_NAME_, config)
-
-    # call the S3Config class
-    s3_obj = S3Config(config)
-
-    _logger.info('start-{}'.format(config.timestamp()))
+    logger = get_pbench_logger(_NAME_, config)
 
-    # add a BACKUP field to the config object
+    # Add the BACKUP and QDIR fields to the config object for convenience.
     config.BACKUP = config.conf.get("pbench-server", "pbench-backup-dir")
     config.QDIR = config.get('pbench-server', 'pbench-quarantine-dir')
 
-    sanity_check(s3_obj, config, _logger)
+    sanity_status = sanity_check(config, logger)
+    if sanity_status != 0:
+        return sanity_status
 
-    prog = os.path.basename(sys.argv[0])
+    # call the S3Config class
+    s3_obj = S3Config(config)
+
+    # make sure the S3 bucket exists
+    try:
+        s3_obj.connector.head_bucket(Bucket='{}'.format(s3_obj.bucket_name))
+    except Exception:
+        logger.exception(
+            "Bucket: {} does not exist or you have no access\n".format(s3_obj.bucket_name))
+        s3_obj = None
+
+    logger.info('start-{}'.format(config.timestamp()))
```

Comment:
Why split this out from the other sanity checks?

Reply:
To avoid S3 backup failures from killing NFS backups.

But we need to also prevent the reserver: NFS backup failures from killing the S3 backups.